src/core/thread/mle.cpp

```diff
@@ -1157,37 +1176,61 @@ void Mle::HandleParentRequestTimer(void)
         }
         else
         {
-            switch (mParentRequestMode)
+            if (mReattachState == kReattachActive)
             {
-            case kMleAttachAnyPartition:
-                if (mPreviousPanId != Mac::kPanIdBroadcast)
+                if (mNetif.GetPendingDataset().Restore() == kThreadError_None)
                 {
-                    mMac.SetChannel(mPreviousChannel);
-                    mMac.SetPanId(mPreviousPanId);
-                    mPreviousPanId = Mac::kPanIdBroadcast;
-                    BecomeDetached();
+                    mNetif.GetPendingDataset().ApplyConfiguration();
+                    mReattachState = kReattachPending;
+                    mParentRequestState = kParentRequestStart;
+                    mParentRequestTimer.Start(kParentRequestRouterTimeout);
                 }
-                else if ((mDeviceMode & ModeTlv::kModeFFD) == 0)
+                else
                 {
-                    SendOrphanAnnounce();
-                    BecomeDetached();
+                    mReattachState = kReattachStop;
                 }
-                else if (mMleRouter.BecomeLeader() != kThreadError_None)
+            }
+            else if (mReattachState == kReattachPending)
+            {
+                mReattachState = kReattachStop;
+                mNetif.GetActiveDataset().Restore();
+                mNetif.GetPendingDataset().Set(mNetif.GetPendingDataset().GetLocal());
```

Comment:
What is the purpose of setting the Pending Dataset here?  To restart the Delay Timer?